### PR TITLE
PHOENIX-4731 Make running transactional unit tests for a given provider optional

### PR DESCRIPTION
**Pull request by JamesRTaylor**
Enables turning transaction tests off for Tephra and/or Omid. Please review, @ohadshacham and/or @twdsilva. This is a precursor to support of Omid - no new pom dependencies have been introduced. This also moves us to Tephra 0.15.0 which is required for the TAL changes.

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/execute/PhoenixTxIndexMutationGenerator.java [hunk 1]
```diff
@@ -313,7 +326,18 @@ public int compare(Cell o1, Cell o2) {
     private void generateDeletes(PhoenixIndexMetaData indexMetaData,
             Collection<Pair<Mutation, byte[]>> indexUpdates,
             byte[] attribValue, TxTableState state) throws IOException {
-        Iterable<IndexUpdate> deletes = codec.getIndexDeletes(state, indexMetaData);
+        byte[] regionStartKey = this.regionStartKey;
```
Comment (twdsilva) [inline phoenix-core/src/main/java/org/apache/phoenix/execute/PhoenixTxIndexMutationGenerator.java]: Just for my understanding, why do we now need the start and stop key to generate the index deletes and upserts?

File: phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java [hunk 2]
```diff
@@ -184,9 +184,24 @@ public boolean apply(PTable index) {
...
         List<PTable> indexes = dataTable.getIndexes();
-        serialize(dataTable, ptr, indexes, connection);
+        serializeServerMaintainedIndexes(dataTable, ptr, indexes, connection);
     }
 
+    public static void serializeServerMaintainedIndexes(PTable dataTable, ImmutableBytesWritable ptr,
+            List<PTable> indexes, PhoenixConnection connection) {
+        Iterator<PTable> indexesItr = Collections.emptyListIterator();
+        boolean onlyLocalIndexes = dataTable.isImmutableRows() || dataTable.isTransactional();
+        if (onlyLocalIndexes) {
```
Comment (twdsilva) [inline phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java]: Do we do anything special for transactional tables with local indexes?

Comment (JamesRTaylor) [reply phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java]: Long story short, we've disallowed local indexes on Omid for now due to PHOENIX-4760. Once this is fixed, we can turn it back on. Given that index updates are transactional with data updates,  there's not a great need IMHO for transactional local indexes, though. Having said that, there were a few things we needed to do for local indexes:
* Support client-side index maintenance for local indexes
* Handle the initial population (i.e. adding shadow columns) for a local index on the server side

**Comment by stoty** [comment]
Already merged.

**Pull request closed without merging**